### DISCUSSION TITLE
Add a way to get the vector capacity given `duckdb_vector`

### DISCUSSION THREAD

**Discussion by Swoorup** [duckdb/duckdb#12553] [Ideas]
Why do you want this feature? 

`duckdb_vector_size` only returns the initial size of the vector that is by default of 2048 size.

It appears you can change the [capacity](https://github.com/duckdb/duckdb/issues/12268#issuecomment-2133805908) and go over that limit using [duckdb_data_chunk_set_size](https://duckdb.org/docs/api/c/api#duckdb_data_chunk_set_size)? or [duckdb_list_vector_reserve](https://duckdb.org/docs/api/c/api#duckdb_list_vector_reserve) depending on the parent type.

However there is no way to get the current capacity using only `duckdb_vector` handle. On the client side, you need to keep tracking the length from the places it was allocated like the data chunk or the list vector, which becomes [cumbersome quickly](https://github.com/Swoorup/duckdb-rs/blob/sj-merge-duckdb-fork/crates/duckdb/src/core/vector.rs#L355). In languages like rust, this becomes a problem if you want to ensure you haven't written past the allocated buffer.

**Comment by Tishj** [comment]
Vectors are not aware of their size, that's what the DataChunk is for.
ListVectors are aware of the size of the child buffer, not of their own size, for example:
```
[1, 2, 3]
[4, 5]
```

The size of the vector is 2, that's stored in the DataChunk, the size of the child vector is 5 (`1,2,3,4,5`), that is stored in the ListVector

**Reply by Swoorup** [reply]
Yeah I do get that, except it makes the few things tricky to get right. The problem I am facing is to correctly propagate the capacity, when we are writing to the vector so there is no buffer overruns, like here: 
https://github.com/Swoorup/duckdb-rs/blob/sj-merge-duckdb-fork/crates/duckdb/src/core/vector.rs#L139-L141

Other option is to keep passing the length from higher up to `as_mut_slice` method.

As you said the size depends on the container like ListVector, ArrayVector or DataChunk.

If that isn't workable, even something to the notion of maximum possible length in bytes that be written to `duckdb_vector` would be nicer.

**Reply by Tishj** [reply]
Internally we pass around the size separately when working with Vectors
